### PR TITLE
feat(manifest-v3): Upgrade Application Insights

### PR DESCRIPTION
**Pull request by madalynrose**
#### Details

Use `applicationinsights-web` node module instead of the deprecated `applicationinsights-js` one.

##### Motivation

`applicationinsights-web` should work reliably with service workers as of v2.6.3 (implements a 

### Context
 
This PR updates the current usage of application insights but does not focus on service worker compatibility yet.

#### Pull request checklist

- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.

### CODE REVIEW

File: src/tests/unit/tests/background/telemetry/app-insights-telemetry-client.test.ts [hunk 1]
```diff
@@ -1,24 +1,24 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+import { IConfig, ITelemetryContext, ITelemetryTrace } from '@microsoft/applicationinsights-common';
```
Comment (dbjorge) [inline src/tests/unit/tests/background/telemetry/app-insights-telemetry-client.test.ts:3]: You can avoid the need for the extra `@microsoft/applicationinsights-common` dependency by importing `IConfig` from the re-export in `@microsoft/applicationinsights-web` and replacing the `ITelemetry*` imports with:

```ts
    type ITelemetryContext = Initialization["context"];
    type ITelemetryTrace = ITelemetryContext["telemetryTrace"];
```

File: package.json [hunk 2]
```diff
@@ -170,6 +170,8 @@
         "webpack-node-externals": "^3.0.0"
     },
     "dependencies": {
+        "@microsoft/applicationinsights-common": "^2.7.3",
+        "@microsoft/applicationinsights-web": "^2.7.3",
```
Comment (dbjorge) [inline package.json:174]: Can we drop the old `applicationinsights-js` dependency with this change?

Comment (dbjorge) [reply package.json:174]: (+ `@types/applicationinsights-js`

File: src/tests/unit/tests/background/telemetry/app-insights-telemetry-client.test.ts (hunk 1 above)
Comment (madalynrose) [reply src/tests/unit/tests/background/telemetry/app-insights-telemetry-client.test.ts:3]: ooh nice catch!!

**Comment by madalynrose** [comment]
@dbjorge I finally got around to fixing that null check!

File: src/background/telemetry/app-insights-telemetry-client.ts [hunk 3]
```diff
@@ -61,14 +55,15 @@ export class AppInsightsTelemetryClient implements TelemetryClient {
...
-        this.applicationInsights.addTelemetryInitializer((telemetryItem: ExtendedTelemetryItem) => {
-            const baseData = telemetryItem.data.baseData;
-            baseData.properties = {
-                ...baseData.properties,
-                ...this.coreTelemetryDataFactory.getData(),
-            };
-
-            return true;
+        this.applicationInsights.addTelemetryInitializer((telemetryItem: ITelemetryItem) => {
+            const baseData = telemetryItem.baseData;
```
Comment (dbjorge) [inline src/background/telemetry/app-insights-telemetry-client.ts:59]:
```suggestion
const originalBaseData = telemetryItem?.baseData ?? {};
telemetryItem.baseData = {
    ...originalBaseData,
    ...theFactoryData
};
```

**Review by dbjorge — APPROVED** [review]

**Pull request merged**